src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ConstVal.cs

```diff
@@ -20,149 +21,77 @@ internal enum ConstValKind
         Decimal,
         IntPtr,
         Float,
-        Boolean,
-        Lim
-    };
-
+        Boolean
+    }
 
-    internal sealed class CONSTVAL
+    internal sealed class ConstVal
```

Comment:
Should this be a struct instead of a class?  Then caching instances wouldn't be necessary, assuming it was done for perf reasons.